reference/src/clientimpl/src/internal/ta_client.h

```diff
@@ -70,6 +70,11 @@ extern "C" {
 #define RELEASE_PARAM(param) \
     ta_free_shared_memory(param)
 
+#define TA_PARAM_NULL TEEC_NONE
```

Comment:
I'm curious why these are changed to TEEC_MEMREF type inputs.  Is SA3 no longer passing in anything but pointers?  No more values?

Reply:
Is it better to have a varied # of defines that somewhat match the use cases in OPTEE? TEEC lists 15 types as you have defined in sa_ta_types.h.

Reply:
SA3 actually never passed in values. They were always pointers to structures or arrays. These are macros that allow you to switch between using shared_memory and processor memory. At one point, Realtek told us they needed to use shared_memory, but they never ended up doing that. If a future SOC vendor needs shared_memory then they can switch to those macros by defining the USE_SHARED_MEMORY macro.